platform/pulpcore/app/upload.py

```diff
@@ -0,0 +1,33 @@
+import hashlib
+from django.core.files.uploadhandler import TemporaryFileUploadHandler
+from django.core.files.uploadedfile import TemporaryUploadedFile
+
+
+class PulpTemporaryUploadedFile(TemporaryUploadedFile):
+    """
+    A file uploaded to a temporary location in Pulp.
+    """
+
+    def __init__(self, name, content_type, size, charset, content_type_extra=None):
+        self.hashers = {}
+        for hasher in hashlib.algorithms_guaranteed:
+            self.hashers[hasher] = getattr(hashlib, hasher)()
+        super(PulpTemporaryUploadedFile, self).__init__(name, content_type, size, charset, content_type_extra)
+
+
+class PulpFileUploadHandler(TemporaryFileUploadHandler):
```

Comment:
To be more descriptive, consider something like ``HashingFileUploadHandler`` as the name.